compiler/ballerina-lang/src/main/java/io/ballerina/projects/Module.java

```diff
@@ -120,4 +134,118 @@ public Spliterator spliterator() {
+            this.newDocContext = newDocContext;
+            Map<DocumentId, DocumentContext> srcDocContextMap = copySrcDocsfromOld();
+            Map<DocumentId, DocumentContext> testDocContextMap = copyTestDocsfromOld();
+
+            if (oldModule.moduleContext.srcDocumentIds().contains(newDocContext.documentId())) {
+                srcDocContextMap.put(newDocContext.documentId(), newDocContext);
+            } else {
+                testDocContextMap.put(newDocContext.documentId(), newDocContext);
+            }
+            createNewModule(srcDocContextMap, testDocContextMap);
+            return this;
+        }
+
+        /**
+         * Creates a copy of the existing module and adds a new source document to the new module.
+         *
+         * @param documentConfig configurations to create the document
+         * @return an instance of the Module.Modifier
+         */
+        public Modifier addDocument(DocumentConfig documentConfig) {
+            this.newDocContext = DocumentContext.from(documentConfig);
+            Map<DocumentId, DocumentContext> srcDocContextMap = copySrcDocsfromOld();
+            Map<DocumentId, DocumentContext> testDocContextMap = copyTestDocsfromOld();
+
+            srcDocContextMap.put(newDocContext.documentId(), newDocContext);
+            createNewModule(srcDocContextMap, testDocContextMap);
+            return this;
+        }
+
+        /**
+         * Creates a copy of the existing module and adds a new test document to the new module.
+         *
+         * @param documentConfig configurations to create the document
+         * @return an instance of the Module.Modifier
+         */
+        public Modifier addTestDocument(DocumentConfig documentConfig) {
+            this.newDocContext = DocumentContext.from(documentConfig);
+            Map<DocumentId, DocumentContext> srcDocContextMap = copySrcDocsfromOld();
+            Map<DocumentId, DocumentContext> testDocContextMap = copyTestDocsfromOld();
+
+            testDocContextMap.put(newDocContext.documentId(), newDocContext);
+            createNewModule(srcDocContextMap, testDocContextMap);
+            return this;
+        }
+
+        /**
+         * Creates a copy of the existing module and removes the specified document from the new module.
+         *
+         * @param documentId documentId of the document to remove
+         * @return an instance of the Module.Modifier
+         */
+        public Modifier removeDocument(DocumentId documentId) {
+            Map<DocumentId, DocumentContext> srcDocContextMap = copySrcDocsfromOld();
+            Map<DocumentId, DocumentContext> testDocContextMap = copyTestDocsfromOld();
+
+            if (oldModule.moduleContext.srcDocumentIds().contains(documentId)) {
+                srcDocContextMap.remove(documentId);
+            } else {
+                testDocContextMap.remove(documentId);
+            }
+            createNewModule(srcDocContextMap, testDocContextMap);
```

Comment:
Follow method in https://github.com/ballerina-platform/ballerina-lang/pull/26009/files#r493393568

Reply:
Fixed in 9a7d6fe